packages/repository/src/relations/relation.types.ts

```diff
@@ -22,6 +22,13 @@ export interface RelationDefinitionBase {
    */
   type: RelationType;
 
+  /**
+   * True for relations targeting multiple instances (e.g. HasMany),
+   * false for relations with a single target (e.g. BelongsTo, HasOne).
+   * This property is needed by OpenAPI/JSON Schema generator.
+   */
+  targetsMany?: boolean;
```

Comment:
I think that this should be a required property like in the issue description. What do you think?

Reply:
I kept it optional for backward compatibility. But if you think otherwise I can make this as required. Thoughts @bajtos ?

Reply:
That's a good point, let's wait for @bajtos' thoughts, but can you please still add it to the relation definitions? e.g.:

```ts
export interface HasManyDefinition extends RelationDefinitionBase {
  type: RelationType.hasMany;
  targetsMany: true;
```

Reply:
added to relation definitions.

Reply:
> I kept it optional for backward compatibility. 

Good point!

I agree that making `targetsMany` a required property can break existing applications or 3rd-party extensions. On the other hand, I would be surprised if any such app or extension actually did exist.

Here is the important part: what is the expected behavior when `targetsMany` is not set? I don't think it's a good idea to default to `false`.

I am proposing the following:
- Add a comment starting with `TODO(semver-major)` explaining that we should make `targetsMany` required in LB5
- Modify the code reading `targetsMany` to throw a helpful error when `targetsMany` does not exist or is `undefined`.  The most important place to change is in `modelToJsonSchema`, but there may be other places to update too.

Reply:
Preferably add a unit test to verify that `modeltoJsonSchema` can handle missing `targetsMany` and demonstrate what happens in such case.

Reply:
> Modify the code reading targetsMany to throw a helpful error when targetsMany does not exist or is undefined

Don't you think throwing error in this case would also mean we are making it backward incompatible ?

Reply:
As I understand it, the error will be thrown only when `modeltoJsonSchema` is called with the new option `includeRelations`. Since this option did not exist before, there shouldn't be any existing code setting it to `true` and thus there shouldn't be any code affected by the new error being thrown.

Reply:
Hmmm. Got it. Let add some error handling and a test case to verify it as well. This makes sense.

Reply:
Done. Please check now.